plugins/modules/azure_rm_apimanagement.py

```diff
@@ -0,0 +1,661 @@
+                        type='str'
+                    ),
+                    description=dict(
+                        type='str'
+                    ),
+                    versioning_scheme=dict(
+                        type='str',
+                        disposition='versioningScheme',
+                        choices=['Segment',
+                                 'Query',
+                                 'Header']
+                    ),
+                    version_query_name=dict(
+                        type='str',
+                        disposition='versionQueryName'
+                    ),
+                    version_header_name=dict(
+                        type='str',
+                        disposition='versionHeaderName'
+                    )
+                )
+            ),
+            value=dict(
+                type='str',
+                disposition='/properties/*'
+            ),
+            format=dict(
+                type='str',
+                disposition='/properties/*',
+                choices=['wadl-xml',
+                         'wadl-link-json',
+                         'swagger-json',
+                         'swagger-link-json',
+                         'wsdl',
+                         'wsdl-link',
+                         'openapi',
+                         'openapi+json',
+                         'openapi-link']
+            ),
+            wsdl_selector=dict(
+                type='dict',
+                disposition='/properties/wsdlSelector',
+                options=dict(
+                    wsdl_service_name=dict(
+                        type='str',
+                        disposition='wsdlServiceName'
+                    ),
+                    wsdl_endpoint_name=dict(
+                        type='str',
+                        disposition='wsdlEndpointName'
+                    )
+                )
+            ),
+            api_type=dict(
+                type='str',
+                disposition='/properties/apiType',
+                choices=['http', 'soap']
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.service_name = None
+        self.api_id = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.url = None
+        self.status_code = [200, 201, 202]
+        self.to_do = Actions.NoAction
+
+        self.body = {}
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = '2020-06-01-preview'
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        super(AzureApiManagement, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                         supports_check_mode=True,
```

Comment:
519, 520,524, 598, 620 and 639 continuation line under-indented for visual indent